### ISSUE TITLE
Add support for KaTex?

### ISSUE THREAD

**Issue by Geliv** [typora/typora-issues#888]
Hoping typora can add support for `KaTex`, which is the fastest math typesetting library for the web.
And it's also better than mathjax in many other aspects. [(see webpage)](https://khan.github.io/KaTeX/)
![image](https://user-images.githubusercontent.com/20089973/30103913-abca02ee-9326-11e7-986e-cd9cfcf6364a.png)

The most desirable feature for me is the printing quality, which is just the same as `LaTex` (not `svg`) for both `html` and printed `pdf`. 

Any plans for this?
Many thanks!

**Comment by riteme** [comment]
:+1:  +1 for this request. KaTeX (now 0.10.0-beta) has improved a lot so far that many features were accomplished, such as `\tag`, `\newcommand`, etc. Personally speaking, I have decided to replace MathJax with KaTeX in my own blog since there's no error in rendering math formulas with KaTeX anymore. By contrast, though MathJax is more compatible, it's *really* slow especially when math formulas increase, making my browser & Typora so laggy.

Another advantage I observed is KaTeX's pure html output, which is more natural and more customizable. Due to this feature, it even supports breaking in formulas:

![](https://gitee.com/riteme/blogimg/raw/master/other/typora-katex-1.png)

which is impossible for current MathJax (2.7).

I do hope there would be a option for switching between KaTeX and MathJax.

**Comment by abnerlee** [comment]
The missing part is https://github.com/Khan/KaTeX/issues/350 and https://github.com/Khan/KaTeX/issues/50

**Comment by Foadsf** [comment]
It would be great if Typora could support the ```katex ``` and ```math ``` syntax.

**Comment by peetteerr** [comment]
Have to admit there is indeed a big flaw that Typora's math rendering performance is rapidly becoming sluggish when having writing a few lines of formulae.

**Comment by Ramesh-X** [comment]
Any update on this issue..?

**Comment by Rullec** [comment]
Support, it would be awesome to support Katex.

**Comment by ice1000** [comment]
> The missing part is [KaTeX/KaTeX#350](https://github.com/KaTeX/KaTeX/issues/350) and [KaTeX/KaTeX#50](https://github.com/KaTeX/KaTeX/issues/50)

Both issues are fixed now 🤔

**Comment by guesswh0** [comment]
> > The missing part is [KaTeX/KaTeX#350](https://github.com/KaTeX/KaTeX/issues/350) and [KaTeX/KaTeX#50](https://github.com/KaTeX/KaTeX/issues/50)
> 
> Both issues are fixed now 🤔

It would be nice to see both supported

**Comment by abnerlee** [comment]
KaTeX seems great now,  now remaining issue should be https://github.com/KaTeX/KaTeX/issues/375

If we use dom / HTML output, then the exported HTML / epub may requires [font](https://katex.org/docs/font.html) to be somewhere so other people can view the exported HTML normally.

But we will move to mathjax 3 for faster speed for now.

**Comment by xujiujiu-9** [comment]
Yes, It would be nice if users can change between Mathjax and Katex in the settings.

**Comment by AhmedThahir** [comment]
Any updates?

**Comment by ice1000** [comment]
> Any updates?

The blocking issue is still blocking.

**Comment by xingyuXXX** [comment]
need katex really

**Comment by zhuozhiyongde** [comment]
@abnerlee Hope there is a KaTeX option.
